cmd/snap-confine/ns-support.c

```diff
@@ -406,19 +459,25 @@ static int sc_inspect_and_maybe_discard_stale_ns(int mnt_fd,
 		die("support process for mount namespace inspection exited abnormally");
 	}
 	// If the namespace is up-to-date then we are done.
-	if (value == SC_DISCARD_NO) {
-		debug("preserved mount namespace can be reused");
-		return 0;
-	}
-	// The namespace is stale, let's check if we can discard it.
-	if (sc_cgroup_freezer_occupied(inv->snap_instance)) {
-		// Some processes are still using the namespace so we cannot discard it
-		// as that would fracture the view that the set of processes inside
-		// have on what is mounted.
-		debug("preserved mount namespace is stale but occupied");
+	switch (value) {
+	case SC_DISCARD_NO:
+		debug("preserved mount is not stale, reusing");
 		return 0;
+	case SC_DISCARD_SHOULD:
```

Comment:
`SC_DISCARD_MAYBE`, `SC_DISCARD_TRY`?

Reply:
Honestly I find the original names more descriptive.